.pre-commit-config.yaml

```
@@ -0,0 +1,27 @@
+# pre-commit install
+
+repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: f71fa2c1f9cf5cb705f73dffe4b21f7c61470ba9 # v4.4.0
+    hooks:
+      - id: trailing-whitespace
+      - id: end-of-file-fixer
+      - id: debug-statements
+      - id: check-ast
+      - id: mixed-line-ending
+      - id: check-yaml
+        args: [--allow-multiple-documents]
+      - id: check-added-large-files
+
+  - repo: https://github.com/pre-commit/mirrors-prettier
+    rev: fc260393cc4ec09f8fc0a5ba4437f481c8b55dc1 # frozen: v3.0.3
+    hooks:
+      - id: prettier
+        files: \.(md|rst|toml|yml|yaml)
+        args: [--prose-wrap=preserve]
+
+  - repo: https://github.com/charliermarsh/ruff-pre-commit
+    rev: 6fcbf19c7cf351a8dc3517a0fed3ebfa05610de2 # frozen: v0.0.290
```

This is using an old repo that is not kept up-to-date.

```suggestion
  - repo: https://github.com/astral-sh/ruff-pre-commit
    rev: v0.1.9
```

Any reason to use hash instead of tag? If so, then:

```suggestion
  - repo: https://github.com/astral-sh/ruff-pre-commit
    rev: 7bcc70ca475b87e0fdee2511300c74b25babe0b3  # frozen: v0.1.9
```